extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcContextPropagationConfig.java

```diff
@@ -0,0 +1,16 @@
+package io.quarkus.arc.deployment;
+
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigItem;
+
+@ConfigGroup
+public class ArcContextPropagationConfig {
+
+    /**
+     * If set to true and SmallRye Context Propagation extension is present then enable the context propagation for CDI
+     * contexts.
```

Comment:
As a user - what happens if I set it to true but the extension is not present?

Reply:
Noop.

Reply:
The description is IMO very clear that context propagation for CDI contexts is only enabled if both conditions are met.

Reply:
Yes it's clear what is required to activate it, but it's unclear what happens in the other conditions.

I wonder if we shouldn't have an exception if someone is activating it, but the extension is missing.

Reply:
> Yes it's clear what is required to activate it, but it's unclear what happens in the other conditions.

Well, I think it is clear - the context propagation is disabled but I get your point. The thing is that we cannot fail the build as long as `true` is the default value.

Reply:
> Yes it's clear what is required to activate it, but it's unclear what happens in the other conditions.
> 
> I wonder if we shouldn't have an exception if someone is activating it, but the extension is missing.

It is `true` by default, so most users are "trying to enable it" without even knowing it and we cannot throw an exception in such scenario.
The only other scenario is trying to disable it when CP extension is missing in which case it wouldn't propagate CDI anyway.

Reply:
I think you could reword it slightly to avoid possible confusion. Something like:

> If set to true, enables propagation of CDI contexts. Only makes sense when the SmallRye Context Propagation extension is present. When it is not, this configuration is option meaningless.

Reply:
That works for me :+1: as a reasonable compromise. I'd still argue that an exception is better: if someone enables it but forgets to add the extension (or attempts to add it but something goes wrong in doing so), they would appreciate to be informed.